experimental/adaptive-tool/src/lg/providers/completion.ts

```diff
@@ -141,6 +165,70 @@ class LGCompletionItemProvider implements vscode.CompletionItemProvider {
             }
             return {isInStruct:false, struType:undefined};
     }
-}
 
+    AddToCompletion(options: { [key: string] : { [key: string] : string; }; }, items: vscode.CompletionItem[]) {
+        for (let option in options) {
+            let completionItem = new vscode.CompletionItem(" " + option);
+            completionItem.detail = options[option]["detail"];
+            completionItem.documentation = options[option]["documentation"];
+            completionItem.insertText = options[option]["insertText"];
+            items.push(completionItem);
+        }
+    }
+}
 
+const optionsMap : { [key: string] : { [key: string] : { [key: string] : string; }; }; } = {
```

Comment:
there is no need to construct a strong type, a simple object is enough.
```
const optionsMap : { [key: string] : { [key: string] : { [key: string] : string; }; }; } = {"a": {"b":"c"}}
- >
 const optionsMap  = {a: {b: 'c'}}
```

So the scope can be instead as:
```
const optionsMap = {
    options: { 
        "@strict": {
            detail: " @strict = true",
            documentation: "Developers who do not want to allow a null evaluated result can implement the strict option.", 
            insertText: " @strict"},
      xxxx
    },
    strictOptions : {
        true: {
            detail: " true",
            documentation: "Null error will throw a friendly message.",
            insertText: " true"
        },
        false: {
            detail: " false",
            documentation: "A compatible result will be given.",
            insertText: " false"
        }
    },
    xxxx
}
```
This is more readable